lib/_stream_writable.js

```diff
@@ -741,7 +747,7 @@ ObjectDefineProperties(Writable.prototype, {
       const w = this._writableState;
       if (!w) return false;
       if (w.writable !== undefined) return w.writable && !w.ended;
-      return Boolean(!w.destroyed && !w.errored && !w.ending);
+      return Boolean(!w.destroyed && !w.errored && !w.ending && !w.ended);
```

Comment:
There is a "weird" test that sets `ended` explicitly. Very unlikely to have any effect in real code.